Source/WebKit/Platform/IPC/ArgumentCoders.h

```diff
@@ -575,6 +608,14 @@ template<typename KeyArg, typename HashArg, typename KeyTraitsArg> struct Argume
         }
     }
 
+    template<typename Encoder>
+    static void encode(Encoder& encoder, HashCountedSetType&& hashCountedSet)
```

Comment:
It would be nice if we could use consistent styling.  Now sometimes we have 2 encoders, and sometimes we have an encoder that takes a universal reference.

Reply:
It depends on how the move is achievable. For instance, with `std::pair` the single `encode()` that utilizes the forwarding reference works fine because of the `std::get<>` overloads. `WTF::KeyValuePair` doesn't have such luck, so a separate `encode()` overload is used to move out the key and the value.

It's probably possible to get this working even for things like `WTF::KeyValuePair`, i.e. translate the value category of the forwarding reference into an appropriate `std::forward<>` call on the contained objects, I just haven't researched it yet.

Reply:
When moving out the encoded-object's members, the forwarding reference's value category can be applied simply through `std::forward<T>(object).key`. But this doesn't work for containers like vectors or hash maps.

What's necessary there is `std::forward_like`. It's only available in C++23, but we can carry an implementation in `StdLibExtras.h`.
https://en.cppreference.com/w/cpp/utility/forward_like